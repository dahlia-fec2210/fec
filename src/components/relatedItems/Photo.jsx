import React from 'react';
import { RotatingLines } from 'react-loader-spinner';

function Photo({ product, productData }) {
  if (productData) {
    return (
      <img className="related-photo" src={productData[product].photo || 'https://img.ltwebstatic.com/images3_pi/2022/04/06/16492430704a5786a3329d6838490cfcc903aa6996_thumbnail_600x.webp'} alt={product} />
    );
  }
  return (
    <RotatingLines
      strokeColor="grey"
      strokeWidth="5"
      animationDuration="0.75"
      width="96"
      visible
    />
  );
}

export default Photo;
