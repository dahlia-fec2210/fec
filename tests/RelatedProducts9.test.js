/* eslint-disable no-undef */
/* eslint-disable import/extensions */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';

import Photo from '../src/components/relatedItems/Photo.jsx';

function changeProduct(event) {
  event.preventDefault();
  setCurrentProduct(product);
  axios.post(`${serverRoute}/interactions`, { element: `related-product-change-button:${product}`, widget: 'Related Products', time: new Date().toTimeString() });
}

const productData = {
  photos: ['https://images.unsplash.com/photo-1551489186-cf8726f514f8?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1650&q=80'],
  thumbnails: ['https://images.unsplash.com/photo-1551489186-cf8726f514f8?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80'],
  price: 900.00,
  salePrice: null,
};

test('renders photo for outfit list card with product 37315', async () => {
  const tree = renderer
    .create(<Photo product={37311} productData={productData} changeProduct={changeProduct} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
