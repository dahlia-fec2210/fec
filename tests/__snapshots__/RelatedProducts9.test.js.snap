// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders photo for outfit list card with product 37315 1`] = `
<div
  className="related-photo"
>
  <img
    alt={37311}
    className="related-photo"
    src="https://images.unsplash.com/photo-1551489186-cf8726f514f8?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1650&q=80"
  />
  <div
    className="related-thumbnails"
    onClick={[Function]}
  >
    <img
      alt="thumbnail"
      data-testid="thumbnail"
      onClick={[Function]}
      src="https://images.unsplash.com/photo-1551489186-cf8726f514f8?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80"
      style={
        {
          "transform": "translate(-0px, 0)",
          "transition": "transform 600ms ease-in-out",
        }
      }
    />
  </div>
  <div
    className="mini-left-arrow"
    onClick={[Function]}
  />
  <div
    className="mini-right-arrow"
    data-testid="mini-right-arrow"
    onClick={[Function]}
  />
</div>
`;
